src/pages/course.js

```diff
@@ -32,6 +33,23 @@ const Course = props => {
     })
   }, [props.courseId])
 
+  const handleFetchQueueData = () => {
```

Comment:
If you use the `Content-Disposition` method mentioned in my other comment, you should be able to avoid doing all this stuff on the client and let the browser handle downloading for you.